### PR TITLE
fix(polymorphism): fix regression introduced with ember-data-model-fragments 6.0.0 where owner is not passed to typeKey function

### PR DESCRIPTION
**Pull request by VincentMolinie**
Will also:
- fix adopted-ember-addons/ember-data-model-fragments#470

### CODE REVIEW

File: addon/record-data.js [hunk 1]
```diff
@@ -715,12 +717,24 @@ export default class FragmentRecordData extends RecordData {
         // strip fragments from the attributes so the super call does not process them
         delete data.attributes[key];
 
+        subFragmentsToProcess.push({ key, behavior, canonical, attributes });
+      }
+    }
+
+    // Wee need first the attributes to be setupd before the fragment to be able to access them (for polymorphic fragments for example)
```
Comment (VincentMolinie) [inline addon/record-data.js:724]:
```suggestion
    // Wee need first the attributes to be setup before the fragment, to be able to access them (for polymorphic fragments for example)
```

File: addon/record-data.js [hunk 2]
```diff
@@ -732,13 +746,12 @@ export default class FragmentRecordData extends RecordData {
...
     }
 
-    const changedAttributeKeys = super.pushData(data, calculateChange);
     const changedKeys = mergeArrays(changedAttributeKeys, changedFragmentKeys);
     if (gte('ember-data', '4.5.0') && changedKeys?.length > 0) {
       internalModelFor(this).notifyAttributes(changedKeys);
     }
     // on ember-data 2.8 - 4.4, InternalModel.setupData will notify
-    return changedKeys;
+    return changedKeys || [];
```
Comment (VincentMolinie) [inline addon/record-data.js:754]: This line fix this issue https://github.com/adopted-ember-addons/ember-data-model-fragments/issues/470

Comment (knownasilya) [reply addon/record-data.js:754]: Should this be a `??`?

Comment (VincentMolinie) [reply addon/record-data.js:754]: This methods should always return an array as it is actually doing a `changedKeys.length` afterward. So I prefer to be safe

**Review by knownasilya — APPROVED** [review]

**Pull request merged**